packages/twenty-website/src/app/user-guide/[slug]/page.tsx

```diff
@@ -1,15 +1,23 @@
+'use client';
+import { use } from 'react';
+
 import UserGuideContent from '@/app/components/user-guide/UserGuideContent';
 import { getPost } from '@/app/get-posts';
+import { DeviceType, useDeviceType } from '@/app/ui/utilities/useDeviceType';
 
-export default async function UserGuideSlug({
+export default function UserGuideSlug({
   params,
 }: {
   params: { slug: string };
 }) {
   const basePath = '/src/content/user-guide';
-  const mainPost = await getPost(
-    params.slug && params.slug.length ? params.slug : 'home',
-    basePath,
+  const deviceType = useDeviceType();
+  const mainPost = use(
+    getPost(
+      params.slug && params.slug.length ? params.slug : 'home',
+      basePath,
+      deviceType === DeviceType.DESKTOP,
```

Comment:
I would use CSS / frontend code to hide the TOC instead of trying to do it on the server